src/shared/dlt_user_shared.c

```diff
@@ -175,4 +198,26 @@ DltReturnValue dlt_user_log_out3(int handle, void *ptr1, size_t len1, void *ptr2
     }
 
     return DLT_RETURN_OK;
-}
\ No newline at end of file
+}
+
+DltReturnValue dlt_user_log_out3_with_timeout(int handle, void *ptr1, size_t len1, void *ptr2, size_t len2, void *ptr3, size_t len3)
+{
+    if (handle < 0)
+        /* Invalid handle */
+        return DLT_RETURN_ERROR;
+
+    fd_set fds;
+    FD_ZERO(&fds);
+    FD_SET(handle, &fds);
+
+    struct timeval tv = { 0, 100000000 };
```

Comment:
100000000 usec = 100 sec.
Isn't this a bit to much?
Please add a configurable value for all these values. Is 10 seconds sufficient as default value? This would be more readable:
struct timeval tv = { 10, 0 };

Reply:
Good catch, I checked our internal fork and I somehow screwed up while applying this patch here. We're actually using 1 second timeout and not 100. 
I adjusted the value to reflect this. 
Also made it configurable via CMake